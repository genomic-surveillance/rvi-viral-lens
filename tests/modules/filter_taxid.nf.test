nextflow_process {

    name "Test Process filter_taxids"
    script "modules/filter_taxid.nf"
    process "filter_taxids"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                // outdir = "tests/results"
                kraken_report="${projectDir}/tests/test_data/test_kraken_databases/test.report.txt"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(params.meta, file(params.kraken_report))
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
