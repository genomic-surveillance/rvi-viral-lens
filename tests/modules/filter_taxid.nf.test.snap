{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "null",
                        "null",
                        "null"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-02-28T16:53:06.502917"
    }
}